### ISSUE TITLE
Crash opening 1.11-pre1 world

### ISSUE THREAD

**Issue by mattmoss** [toolbox4minecraft/amidst#247]
```
[info] Amidst v4.2-beta2
[info] Current system time: 2016-11-11 13:18:29.263
[info] System.getProperty("os.name") == 'Mac OS X'
[info] System.getProperty("os.version") == '10.10.5'
[info] System.getProperty("os.arch") == 'x86_64'
[info] System.getProperty("java.version") == '1.8.0_91'
[info] System.getProperty("java.vendor") == 'Oracle Corporation'
[info] System.getProperty("sun.arch.data.model") == '64'
[info] Enabling OpenGL.
[info] Biome profile activated: default
[info] using '.minecraft' directory at: '/Users/mattmoss/Library/Application Support/minecraft', libraries: '/Users/mattmoss/Library/Application Support/minecraft/libraries'
[info] using biome profiles at: 'biome'
[info] Scanning for profiles.
[info] Successfully loaded profile list.
[info] Beginning latest version list load.
[info] Attempting to download remote version list...
[info] Successfully loaded version list. URL: https://launchermeta.mojang.com/mc/game/version_manifest.json
[info] using minecraft launcher profile 'mattmoss' with versionId '1.11-pre1'
[info] Loading libraries.
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/mojang/netty/1.6/netty-1.6.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/oshi-project/oshi-core/1.1/oshi-core-1.1.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/net/java/dev/jna/jna/3.4.0/jna-3.4.0.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/net/java/dev/jna/platform/3.4.0/platform-3.4.0.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/ibm/icu/icu4j-core-mojang/51.2/icu4j-core-mojang-51.2.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/net/sf/jopt-simple/jopt-simple/4.6/jopt-simple-4.6.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/paulscode/codecjorbis/20101023/codecjorbis-20101023.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/paulscode/codecwav/20101023/codecwav-20101023.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/paulscode/libraryjavasound/20101123/libraryjavasound-20101123.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/paulscode/librarylwjglopenal/20100824/librarylwjglopenal-20100824.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/paulscode/soundsystem/20120107/soundsystem-20120107.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/io/netty/netty-all/4.0.23.Final/netty-all-4.0.23.Final.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/google/guava/guava/17.0/guava-17.0.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/apache/commons/commons-lang3/3.3.2/commons-lang3-3.3.2.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/commons-io/commons-io/2.4/commons-io-2.4.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/commons-codec/commons-codec/1.9/commons-codec-1.9.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/google/code/gson/gson/2.2.4/gson-2.2.4.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/mojang/authlib/1.5.24/authlib-1.5.24.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/com/mojang/realms/1.10.0/realms-1.10.0.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/apache/commons/commons-compress/1.8.1/commons-compress-1.8.1.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/apache/httpcomponents/httpclient/4.3.3/httpclient-4.3.3.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/apache/httpcomponents/httpcore/4.3.2/httpcore-4.3.2.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/it/unimi/dsi/fastutil/7.0.12_mojang/fastutil-7.0.12_mojang.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/apache/logging/log4j/log4j-api/2.0-beta9/log4j-api-2.0-beta9.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/apache/logging/log4j/log4j-core/2.0-beta9/log4j-core-2.0-beta9.jar
[info] Skipping library: org.lwjgl.lwjgl:lwjgl:2.9.4-nightly-20150209
[info] Skipping library: org.lwjgl.lwjgl:lwjgl_util:2.9.4-nightly-20150209
[info] Skipping library: org.lwjgl.lwjgl:lwjgl-platform:2.9.4-nightly-20150209
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl/2.9.2-nightly-20140822/lwjgl-2.9.2-nightly-20140822.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl_util/2.9.2-nightly-20140822/lwjgl_util-2.9.2-nightly-20140822.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl-platform/2.9.2-nightly-20140822/lwjgl-platform-2.9.2-nightly-20140822-natives-osx.jar
[info] Found library: /Users/mattmoss/Library/Application Support/minecraft/libraries/net/java/jinput/jinput-platform/2.0.5/jinput-platform-2.0.5-natives-osx.jar
[info] Unable to recognise Minecraft Version with the magic string "rroumhkfph[Llw;mt[J[[Jmp".
[info] Reading 1.11-pre1.jar
[info] Jar load complete.
[info] Searching for classes...
[info] Found class BlockInit: kt
[info] Found class GenLayer: bah
[info] Found class IntCache: baf
[info] Found class WorldType: ajv
[info] Class search complete.
[info] Loading classes...
[info] unable to find the real class method bah.a(long, @WorldType) -> GenLayer.initializeAllBiomeGenerators2
[info] unable to find the real class method bah.a(long, @WorldType, String) -> GenLayer.initializeAllBiomeGenerators3
[info] Classes loaded.
[info] Minecraft load complete.
[info] Not writing to seed history file, because it does not exist: history.txt
[info] Creating world with seed '-886332549' and type 'Default'
[info] Using the following generator options: 
[crash] Amidst has encounted an uncaught exception on the thread Thread[AWT-EventQueue-0,6,main]
[crash] java.lang.NullPointerException
	at amidst.clazz.symbolic.SymbolicClass.callStaticMethod(SymbolicClass.java:74)
	at amidst.mojangapi.minecraftinterface.local.LocalMinecraftInterface.getGenLayers(LocalMinecraftInterface.java:104)
	at amidst.mojangapi.minecraftinterface.local.LocalMinecraftInterface.createWorld(LocalMinecraftInterface.java:78)
	at amidst.mojangapi.world.WorldBuilder.create(WorldBuilder.java:97)
	at amidst.mojangapi.world.WorldBuilder.fromSaveGame(WorldBuilder.java:73)
	at amidst.mojangapi.MojangApi.createWorldFromSaveGame(MojangApi.java:133)
	at amidst.gui.main.Actions.openSaveGame(Actions.java:92)
	at amidst.gui.main.menu.Menus.lambda$item$55(Menus.java:131)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:833)
	at com.apple.laf.AquaMenuItemUI.doClick(AquaMenuItemUI.java:157)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:877)
	at java.awt.Component.processMouseEvent(Component.java:6535)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6300)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

```

**Comment by stefandollase** [comment]
Thanks for the report. This indeed seems to be an issue with Minecraft `1.11-pre1`.

**Comment by MattSturgeon** [comment]
I get the same error on the [final release](https://launcher.mojang.com/mc/game/1.11/client/780e46b3a96091a7f42c028c615af45974629072/client.jar) of 1.11 using AMIDST 4.1 (not tested the git version yet)

**Comment by stefandollase** [comment]
The bug is not fixed in the latest git version. We still have to figure out the cause (Mojang probably changed one of the classes that we use) and how to fix it (this will be more difficult).

**Comment by Mendrak** [comment]
Not sure if this helps but I can load a 16w20a profile and load my 1.11 world.

**Comment by stefandollase** [comment]
@Mendrak
Thanks for letting us know. Indeed, the crash is caused by using a 1.11 Minecraft profile, not by loading a world that was created by a 1.11 Minecraft profile.

**Comment by Vmachine1976** [comment]
You can edit your minecraft profile to stay in the 1.10.2 then Amidst is working (And wait for and 1.11 fix)

**Comment by stefandollase** [comment]
@Vmachine1976
This is correct. You can also create a new Minecraft profile with the old Minecraft version for Amidst. This allows you to play Minecraft 1.11 with your current Minecraft profile and to use Amidst with the new Minecraft profile.

**Comment by mattmoss** [comment]
Thanks for the ideas; I'll wait until Amidst is working.

**Comment by moulins** [comment]
I've looked at the crash in more details, and here's what I found:

As the stacktrace says, the crash happens in ```LocalMinecraftInterface.getGenLayers```, while calling the minecraft method which initializes the biome objects. In minecraft 1.11, the code tries to call the method ```initializeAllBiomeGenerators2```, which doesn't exist. In minecraft 1.10, ```initializeAllBiomeGenerators3``` is called instead, which suggests that this method can't be found anymore in minecraft 1.11.

After decompiling the relevant class of MC1.11, I found that the method is still there, but its signature has changed: it is now
```java
public static bah[] a(long seed, ajv worldType, avo obj)
// this argument was a String in 1.10 ---------------^
```
After some more looking around, it appears that this ```avo``` object represents the generator settings (```ChunkProviderSettings``` in MCP). Fortunately, Minecraft uses Gson to populate the settings, so we may be able to create an instance of this class relatively easily.

**Comment by stefandollase** [comment]
@moulins
Thanks for starting to investigate the issue. This already does help. The next step is to (1) identify a way to construct an instance of the class `avo` and (2) find a fingerprint to reliably identify the class `avo`, because it is likely to change its name in upcoming versions of the game. What do you think is a stable way to identify the class? Maybe a unique constant that is declared in that class (I did not find any)?  The number of fields seems to be relatively high, so it might be likely to change in the future.

**Comment by stefandollase** [comment]
This is fixed by #269.

**Comment by stefandollase** [comment]
I just released [Amidst v4.2-beta3](https://github.com/toolbox4minecraft/amidst/releases/tag/v4.2-beta3). Please try it out. If there are no major issues I will release Amidst v4.2 this week.
